src/lavinmq/amqp/stream/filters/gis.cr

```diff
@@ -0,0 +1,279 @@
+        raise ArgumentError.new("Radius must be positive") if @radius_km <= 0
+      end
+
+      def match?(headers : AMQP::Table) : Bool
+        msg_point = Point.from_headers(headers)
+        return false unless msg_point
+
+        center.distance_to(msg_point) <= radius_km
+      end
+
+      def to_s(io : IO)
+        io << "RadiusFilter(center=" << @center << ", radius_km=" << @radius_km << ")"
+      end
+    end
+
+    class BoundingBoxFilter
+      include StreamFilter
+
+      getter bbox : BoundingBox
+
+      def initialize(@bbox : BoundingBox)
+      end
+
+      def match?(headers : AMQP::Table) : Bool
+        msg_point = Point.from_headers(headers)
+        return false unless msg_point
+
+        bbox.contains?(msg_point)
+      end
+
+      def to_s(io : IO)
+        io << "BoundingBoxFilter(" << @bbox << ")"
+      end
+    end
+
+    # Represents a bounding box defined by min/max latitude and longitude
+    struct BoundingBox
+      getter min_lat : Float64
+      getter max_lat : Float64
+      getter min_lon : Float64
+      getter max_lon : Float64
+
+      def initialize(@min_lat : Float64, @max_lat : Float64, @min_lon : Float64, @max_lon : Float64)
+        validate!
+      end
+
+      private def validate!
+        raise ArgumentError.new("min_lat must be <= max_lat") unless @min_lat <= @max_lat
+        raise ArgumentError.new("min_lon must be <= max_lon") unless @min_lon <= @max_lon
+        raise ArgumentError.new("Latitude must be between -90 and 90") unless @min_lat >= -90 && @max_lat <= 90
+        raise ArgumentError.new("Longitude must be between -180 and 180") unless @min_lon >= -180 && @max_lon <= 180
+      end
+
+      def contains?(point : Point) : Bool
+        point.lat >= @min_lat && point.lat <= @max_lat &&
+          point.lon >= @min_lon && point.lon <= @max_lon
+      end
+
+      def to_s(io : IO)
+        io << "BoundingBox(lat=" << @min_lat << ".." << @max_lat
+        io << ", lon=" << @min_lon << ".." << @max_lon << ")"
+      end
+    end
+
+    # Represents a polygon as an array of points
+    struct Polygon
+      getter points : Array(Point)
+
+      def initialize(@points : Array(Point))
+        raise ArgumentError.new("Polygon must have at least 3 points") if @points.size < 3
+      end
+
+      # Point-in-polygon test using ray casting algorithm
+      # https://en.wikipedia.org/wiki/Point_in_polygon
+      def contains?(point : Point) : Bool
+        inside = false
+        j = @points.size - 1
+
+        @points.size.times do |i|
+          pi = @points[i]
+          pj = @points[j]
+
+          # Check if the edge crosses the ray from point to the right
+          # The first condition ensures pj.lon != pi.lon (no vertical edges)
+          if ((pi.lon > point.lon) != (pj.lon > point.lon)) &&
+             (point.lat < (pj.lat - pi.lat) * (point.lon - pi.lon) / (pj.lon - pi.lon) + pi.lat)
+            inside = !inside
+          end
+
+          j = i
+        end
+
```

Comment:
just nitpick, but it would be nice if we did not have this mny spaces between the code lines, its a bit harder to read 👼